rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -1363,29 +1363,30 @@ public void visitDeclarationExpression(DeclarationExpression expression) {
             List<J.Annotation> leadingAnnotations = visitAndGetAnnotations(expression, classVisitor);
             Optional<MultiVariable> multiVariable = maybeMultiVariable();
             List<J.Modifier> modifiers = getModifiers();
-            TypeTree typeExpr = visitVariableExpressionType(expression.getVariableExpression());
-
-            J.VariableDeclarations.NamedVariable namedVariable;
+            TypeTree typeExpr;
+            List<J.VariableDeclarations.NamedVariable> namedVariables;
             if (expression.isMultipleAssignmentDeclaration()) {
-                // def (a, b) = [1, 2]
-                throw new UnsupportedOperationException("Parsing multiple assignment (e.g.: def (a, b) = [1, 2]) is not implemented");
+                Space prefixBeforeOpenParentheses = whitespace();
+                typeExpr = visitTupleExpressionType(expression.getTupleExpression());
+                JContainer<J.Identifier> identifiers = visit(expression.getTupleExpression());
+                namedVariables = identifiers.getElements().stream()
+                        .map(this::createNamedVariable)
+                        .collect(toList());
+                int first = 0, last = namedVariables.size() - 1;
+                namedVariables.set(first, namedVariables.get(first).withPrefix(prefixBeforeOpenParentheses).withMarkers(Markers.build(singletonList(new OpenParentheses(randomId())))));
+                namedVariables.set(last, namedVariables.get(last).withMarkers(Markers.build(singletonList(new CloseParentheses(randomId())))));
```

Comment:
This is usually a `JContainer` that holds the named variables, then wrapped inside of a `J.Parentheses` instead. This then means that the markers wouldn't be necessary.

Reply:
But it's not including parenthesis, it's just returning `JRightPadded<J.Identifier>`

Reply:
Yes, originally I was talking in more abstract senses.

Specifically for this case, you're getting the `JContainer` back from visiting the identifiers. That is then needing to go directly into `J.NamedVariables`. So in this case, you'd then wrap the `J.NamedVariables` -- if parentheses were present -- into a `J.Parentheses` and return that. The parent element accepts any `Expression` type on the left hand side of the `J.Assignment` that is being built and `J.Parentheses` would meet that constraint.

Reply:
This was more in general. The way to add parentheses around an element is with `J.Parentheses` and when that element is a list it is typically wrapped in a `JContainer`.

I'm not sure what elements you currently have in scope from the Groovy AST, but I'd expect that something similar would be the case here.

Reply:
It's using the same `J.VariableDeclarations` for normal variable declarations. This will require changes to the `J.VariableDeclarations` class. I'm not entirely sure about this approach, though.

Reply:
Perhaps indeed best not to make any structural changes there, as it would mean we'd have to handle that separately downstream when the left hand side can be an instance of J.Parentheses. I'd say a decent trade off to use markers here.

Reply:
I don't think this necessarily needs to be a part of this PR in particular, but this choice here does set a precedent into motion that has a pretty large ripple effect.

As a more specific example, Groovy, Kotlin, C#, Python, and JS (at least) all support destructing-like syntaxes. So by choosing markers here, we too are also likely choosing markers for all of those other languages as well.

To compound on this, the Java community periodically has discussions about adding destructuring support natively, but none of those discussions have yet to raise an accepted JEP.

@sambsnyd or @jkschneider, would either of you want to weigh in here as well?